app/plugins/ProductionModule.java

```diff
@@ -1,75 +1,25 @@
-import com.typesafe.config.Config;
-import com.typesafe.config.ConfigFactory;
-import io.sphere.sdk.categories.Category;
 import io.sphere.sdk.categories.CategoryTree;
-import io.sphere.sdk.categories.queries.CategoryQuery;
 import io.sphere.sdk.client.PlayJavaSphereClient;
-import io.sphere.sdk.client.SphereClient;
-import io.sphere.sdk.client.SphereClientFactory;
-import io.sphere.sdk.queries.PagedQueryResult;
-import play.Application;
-import play.Configuration;
-import play.Logger;
 
-import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.concurrent.TimeUnit;
+
+import play.api.*;
+import play.api.inject.*;
+import scala.collection.Seq;
 
 /**
  * Configuration for the Guice {@link com.google.inject.Injector} which
  * shall be used in production and integration tests.
  */
-public class ProductionModule extends AbstractModule {
-    private final Application app;
-
-    public ProductionModule(Application app) {
-        this.app = app;
-    }
+public class ProductionModule extends Module {
 
     @Override
-    protected void configure() {
-        bind(PlayJavaSphereClient.class).toProvider(PlayJavaSphereClientProvider.class).in(Singleton.class);
-        bind(CategoryTree.class).toProvider(CategoryTreeProvider.class).in(Singleton.class);
-        bind(Configuration.class).toInstance(app.configuration());
-    }
-
-    @Singleton
-    private static class CategoryTreeProvider implements Provider<CategoryTree> {
-        private final PlayJavaSphereClient client;
-
-        @Inject
-        private CategoryTreeProvider(final PlayJavaSphereClient client) {
-            this.client = client;
-        }
-
-        @Override
-        public CategoryTree get() {
-            Logger.info("execute CategoryTreeProvider.get()");
-            final PagedQueryResult<Category> pagedQueryResult = client.execute(CategoryQuery.of()).get(2000, TimeUnit.MILLISECONDS);//TODO this will be most likely moved to a plugin
-            return CategoryTree.of(pagedQueryResult.getResults());
-        }
-    }
-
-
-    @Singleton
-    private static class PlayJavaSphereClientProvider implements Provider<PlayJavaSphereClient> {
-        @Override
-        public PlayJavaSphereClient get() {
-            return createClient();
-        }
-    }
-
-    public static PlayJavaSphereClient createClient() {
-        Logger.info("execute PlayJavaSphereClientProvider.get()");
-        final Config classpathConf = ConfigFactory.load();
-        final String project = classpathConf.getString("sphere.project");
-        final String clientId = classpathConf.getString("sphere.clientId");
-        final String clientSecret = classpathConf.getString("sphere.clientSecret");
-        final SphereClient sphereClient = SphereClientFactory.of().createClient(project, clientId, clientSecret);
-        return PlayJavaSphereClient.of(sphereClient);
+    public Seq<Binding<?>> bindings(Environment environment, play.api.Configuration configuration) {
```

Comment:
Since you are using the Scala API everywhere in this class, I didn't see it necessary. But I guess you want to really make the developer clear it is not the Java API, in which case is good ;)